internal/pkg/db/db.go

```diff
@@ -93,14 +101,81 @@ func (backend *Backend) AddDocument(document *sbom.Document) error {
 func (backend *Backend) GetDocumentByID(id string) (*sbom.Document, error) {
 	document, err := backend.Retrieve(id, nil)
 	if err != nil {
-		backend.Logger.Warn("Document could not be retrieved", "id", id, "err", err)
+		backend.Logger.Debug("Document could not be retrieved", "id", id, "err", err)
 
 		return nil, fmt.Errorf("failed to retrieve document: %w", err)
 	}
 
 	return document, nil
 }
 
+func (backend *Backend) GetDocumentByIDOrAlias(id string) (*sbom.Document, error) {
+	document, idErr := backend.GetDocumentByID(id)
+
+	if document == nil {
+		documents, aliasErr := backend.GetDocumentsByAnnotation(BomctlAnnotationAlias, id)
+		if aliasErr != nil || len(documents) == 0 {
+			backend.Logger.Debug("Document could not be retrieved", "alias", id, "idErr", idErr, "aliasErr", aliasErr)
+
+			return nil, fmt.Errorf("%w", errDocumentAnnotationNotFound)
+		}
+
+		document = documents[0]
+	}
+
+	return document, nil
+}
+
+func (backend *Backend) GetDocumentsByIDOrAlias(ids ...string) ([]*sbom.Document, error) {
+	if len(ids) == 0 {
+		documents, err := backend.GetDocumentsByID()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get documents by ID: %w", err)
+		}
+
+		return documents, nil
+	}
+
+	documents := make([]*sbom.Document, len(ids))
+
+	for idx := range ids {
+		document, err := backend.GetDocumentByIDOrAlias(ids[idx])
+		if errors.Is(err, errDocumentAnnotationNotFound) {
+			return []*sbom.Document{}, nil
+		} else if err != nil {
+			return nil, err
+		}
```

Comment:
Should this really return an error? Or just return an empty list if nothing is found? Trying to mirror SQL behavior

In any case:

```suggestion
		if err != nil {
			if errors.Is(err, errDocumentAnnotationNotFound) {
				return []*sbom.Document{}, nil
			}

			return nil, err
		}
```